File: AtCoder/Browns/python/ARC051-A-Nurie.py
Y, N = "YES", "NO"

x1, y1, r = map(int, input().split())
x2, y2, x3, y3 = map(int, input().split())
d1 = (x2 - x1) ** 2 + (y2 - y1) ** 2 - r ** 2
d2 = (x2 - x1) ** 2 + (y3 - y1) ** 2 - r ** 2
d3 = (x3 - x1) ** 2 + (y2 - y1) ** 2 - r ** 2
d4 = (x3 - x1) ** 2 + (y3 - y1) ** 2 - r ** 2

if all(list(map(lambda x: x <= 0, (d1, d2, d3, d4)))):
    red, blue = Y, N
elif x2 <= x1 - r and x1 + r <= x3 and y2 <= y1 - r and y1 + r <= y3:
    red, blue = N, Y
else:
    red, blue = Y, Y
print(red)
print(blue)

File: AtCoder/Browns/cpp/ARC051-A-Nurie.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

const string Y = "YES", N = "NO";

ll X1, Y1, R;
ll X2, Y2, X3, Y3;

int main() {
  cin >> X1 >> Y1 >> R >> X2 >> Y2 >> X3 >> Y3;
  ll d1 = (X2 - X1) * (X2 - X1) + (Y2 - Y1) * (Y2 - Y1) - R * R;
  ll d2 = (X2 - X1) * (X2 - X1) + (Y3 - Y1) * (Y3 - Y1) - R * R;
  ll d3 = (X3 - X1) * (X3 - X1) + (Y2 - Y1) * (Y2 - Y1) - R * R;
  ll d4 = (X3 - X1) * (X3 - X1) + (Y3 - Y1) * (Y3 - Y1) - R * R;
  string red, blue;
  if (d1 <= 0 && d2 <= 0 && d3 <= 0 && d4 <= 0)
    red = Y, blue = N;
  else if (X2 <= X1 - R && X1 + R <= X3 && Y2 <= Y1 - R && Y1 + R <= Y3)
    red = N, blue = Y;
  else
    red = Y, blue = Y;
  puts(red.c_str());
  puts(blue.c_str());
}